rero_ils/modules/loans/permissions.py

```diff
@@ -17,74 +17,38 @@
-from rero_ils.modules.patrons.api import current_librarian, current_patrons
-from rero_ils.modules.permissions import RecordPermission
+from invenio_access import action_factory, any_user
+from invenio_records_permissions.generators import Generator
 
+from rero_ils.modules.permissions import \
+    AllowedByActionRestrictByOwnerOrOrganisation, RecordPermissionPolicy
 
-class LoanPermission(RecordPermission):
-    """Loan permissions."""
+# Actions to control Loan policy
+search_action = action_factory('loan-search')
+read_action = action_factory('loan-read')
 
-    @classmethod
-    def list(cls, user, record=None):
-        """List permission check.
 
-        :param user: Logged user.
-        :param record: Record to check.
-        :return: True is action can be done.
-        """
-        # user must be authenticated
-        return bool(current_patrons or current_librarian)
+class DisallowedIfAnonymized(Generator):
+    """Disallow if the record is anonymized."""
 
-    @classmethod
-    def read(cls, user, record):
-        """Read permission check.
+    def excludes(self, record=None, *args, **kwargs):
+        """Disallows the given action.
 
-        :param user: Logged user.
-        :param record: Record to check.
-        :return: True is action can be done.
+        :param record: the record to check.
+        :param kwargs: extra arguments.
+        :returns: a list of needs to disabled access.
         """
-        # Read is denied for to_anonymize loans.
-        if record.get('to_anonymize'):
-            return False
-        # staff member (lib, sys_lib) can always read loans
-        if current_librarian \
-           and current_librarian.organisation_pid == \
-           Loan(record).organisation_pid:
-            return True
-        # patron can only read their own loans
-        return Loan(
-            record).patron_pid in [ptrn.pid for ptrn in current_patrons]
-
-    @classmethod
-    def create(cls, user, record=None):
-        """Create permission check.
+        return [any_user] if record and record.get('to_anonymize') else []
 
-        :param user: Logged user.
-        :param record: Record to check.
-        :return: True is action can be done.
-        """
-        # deny all
-        return False
 
-    @classmethod
-    def update(cls, user, record):
-        """Update permission check.
+class LoanPermissionPolicy(RecordPermissionPolicy):
+    """Loan Permission Policy used by the CRUD operations."""
 
-        :param user: Logged user.
-        :param record: Record to check.
-        :return: True is action can be done.
-        """
-        # deny all
-        return False
-
-    @classmethod
-    def delete(cls, user, record):
-        """Delete permission check.
-
-        :param user: Logged user.
-        :param record: Record to check.
-        :return: True if action can be done.
-        """
-        # deny all
-        return False
+    can_search = [
+        DisallowedIfAnonymized(),
```

Comment:
I think `DissalowedIfAnonymized` is not relevant here. We always search for a list of records, vers about 1 record. You can remove it for 'can_search'.